Xamarin.Essentials/WebAuthenticator/WebAuthenticator.ios.tvos..cs

```diff
@@ -0,0 +1,179 @@
+
+namespace Xamarin.Essentials
+{
+    public static partial class WebAuthenticator
+    {
+        static TaskCompletionSource<AuthResult> tcsResponse;
+        static UIViewController currentViewController;
+        static Uri redirectUri;
+
+        internal static async Task<AuthResult> PlatformAuthenticateAsync(Uri url, Uri callbackUrl)
+        {
+            tcsResponse = new TaskCompletionSource<AuthResult>();
+            redirectUri = callbackUrl;
+
+            try
+            {
+                var scheme = redirectUri.Scheme;
+
+                if (!VerifyHasUrlSchemeOrDoesntRequire(scheme))
+                {
+                    tcsResponse.TrySetException(new InvalidOperationException("You must register your URL Scheme handler in your app's Info.plist!"));
+                    return await tcsResponse.Task;
+                }
+
+#if __IOS__
+                void AuthSessionCallback(NSUrl cbUrl, NSError error)
+                {
+                    if (error == null)
+                        OpenUrl(cbUrl);
+                    else
+                        tcsResponse.TrySetException(new NSErrorException(error));
+                }
+
+                if (UIDevice.CurrentDevice.CheckSystemVersion(12, 0))
+                {
+                    var was = new ASWebAuthenticationSession(new NSUrl(url.OriginalString), scheme, AuthSessionCallback);
+                    was.PresentationContextProvider = new ContextProvider(Platform.GetCurrentWindow());
+                    was.Start();
+                    return await tcsResponse.Task;
+                }
+
+                if (UIDevice.CurrentDevice.CheckSystemVersion(11, 0))
+                {
+                    var sf = new SFAuthenticationSession(new NSUrl(url.OriginalString), scheme, AuthSessionCallback);
+                    sf.Start();
+                    return await tcsResponse.Task;
+                }
+
+                if (UIDevice.CurrentDevice.CheckSystemVersion(9, 0))
+                {
+                    var controller = new SFSafariViewController(new NSUrl(url.OriginalString), false)
+                    {
+                        Delegate = new NativeSFSafariViewControllerDelegate
+                        {
+                            DidFinishHandler = (svc) =>
+                            {
+                                // Cancel our task if it wasn't already marked as completed
+                                if (!(tcsResponse?.Task?.IsCompleted ?? true))
+                                    tcsResponse.TrySetException(new OperationCanceledException());
+                            }
+                        },
+                    };
+
+                    currentViewController = controller;
+                    await Platform.GetCurrentUIViewController().PresentViewControllerAsync(controller, true);
+                    return await tcsResponse.Task;
+                }
+#endif
+
+                var opened = UIApplication.SharedApplication.OpenUrl(url);
```

Comment:
Since iOS 8 may not be a common test device, this probably needs to be tested at least once.

Reply:
We only support ios 10 +

Reply:
Right, we can skip the iOS9 check here then and move the fallback of launching safari external to just be for tvos